File: clase3/Persona.py
class Persona:

    # Constructor
    def __init__(self, nombre, apellido, edad):
        self.nombre = nombre
        self.apellido = apellido
        self.edad = edad

    # Método para saludar
    def saludar(self):
        print(f"Hola, soy {self.nombre} {self.apellido}.")

    # Método para cumplir años
    def cumplir_anios(self):
        self.edad += 1
        print(f"¡Ahora tengo {self.edad} años!")

    # Método para mostrar los datos
    def mostrar_datos(self):
        print("===== DATOS DE LA PERSONA =====")
        print(f"Nombre   : {self.nombre}")
        print(f"Apellido : {self.apellido}")
        print(f"Edad     : {self.edad}")

File: clase6/comportamiento/Persona.java
package comportamiento;

public class Persona {

    private String nombre;
    private int edad;

    // Atributo estático (compartido por todas las personas)
    private static int cantidadPersonas = 0;

    public Persona(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
        cantidadPersonas++;
    }

    // Método de instancia
    public void saludar() {
        System.out.println("Hola, soy " + nombre);
    }

    // Método de instancia
    public void presentarse() {
        System.out.println("Me llamo " + nombre + " y tengo " + edad + " años.");
    }

       // Método estático
    public static void mostrarCantidadPersonas() {
        System.out.println("Cantidad de personas creadas: " + cantidadPersonas);
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public static int getCantidadPersonas() {
        return cantidadPersonas;
    }

    public static void setCantidadPersonas(int cantidadPersonas) {
        Persona.cantidadPersonas = cantidadPersonas;
    }
    
}
